### PR TITLE
🚿 Convert the last remaining channel type (Android) to a dynamic type

### PR DESCRIPTION
**Pull request by rowanseymour**

### CODE REVIEW

File: temba/api/v2/serializers.py [hunk 1]
```diff
@@ -466,7 +466,7 @@ def get_country(self, obj):
         return str(obj.country) if obj.country else None
 
     def get_device(self, obj):
-        if obj.channel_type != Channel.TYPE_ANDROID:
+        if not obj.is_android():
```
Comment (rowanseymour) [inline temba/api/v2/serializers.py]: we don't have similar methods for other types, but Android channels are handled differently in so many places that it makes sense

File: temba/channels/types/android/views.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+        self.object.claim(org, self.request.user, self.form.cleaned_data["phone_number"])
+        self.object.save()
+
+        # trigger a sync
+        self.object.trigger_sync()
+
+        return super().form_valid(form)
+
+    def derive_org(self):
```
Comment (rowanseymour) [inline temba/channels/types/android/views.py]: still trying to remember why this is here and other channel claim pages don't have it... 🤔

Comment (nicpottier) [reply temba/channels/types/android/views.py]: Looks like ClaimViewMixin has OrgPermsMixin which has a `derive_org` in it but it is slightly different. Main thing I see is checking the sesson for `org_id` which seems a bit odd honestly. I think it is safe to remove.

Comment (rowanseymour) [reply temba/channels/types/android/views.py]: Wondering if maybe in the past there was some other way to get to the claim page.. maybe on the phone itself 🤷‍♂

**Pull request merged**